resourcescaler.go

```diff
@@ -48,147 +47,142 @@ func New(kubeconfigPath string, namespace string) (scaler_types.ResourceScaler,
+	})
 
-		service.Spec.Selector = service.GetLabels()
-		_, err := s.kubeClientSet.CoreV1().Services(s.namespace).Update(service)
-		if err != nil {
-			s.logger.WarnWith("Failure during update of service with selector",
-				"resource_name", string(resource))
-			return errors.Wrap(err, "Failed updating service instance")
-		}
-	}
+	jsonPatchMapper = append(jsonPatchMapper, map[string]string{
+		"op":    "add",
+		"path":  "/status/state",
+		"value": "waitingForProvisioning",
+	})
 
-	// get deployment by resource name
-	deployment, err := s.kubeClientSet.AppsV1beta1().Deployments(s.namespace).Get(string(resource), meta_v1.GetOptions{})
+	return s.patchIguazioTenantAppServiceSets(namespace, jsonPatchMapper)
+}
+
+func (s *AppResourceScaler) patchIguazioTenantAppServiceSets(namespace string, jsonPatchMapper []map[string]string) error {
+	body, err := json.Marshal(jsonPatchMapper)
+	s.logger.DebugWith("Patching iguazio tenant app service sets", "body", string(body))
 	if err != nil {
-		s.logger.WarnWith("Failure during retrieval of deployment", "resource_name", string(resource))
-		return errors.Wrap(err, "Failed getting deployment instance")
+		return errors.Wrap(err, "Could not marshal json patch mapper")
 	}
 
-	// set deployment num of replicas by scaling factor (0/1)
-	int32scaling := int32(scaling)
-	deployment.Spec.Replicas = &int32scaling
-	_, err = s.kubeClientSet.AppsV1beta1().Deployments(s.namespace).Update(deployment)
+	absPath := []string{"apis", "iguazio.com", "v1beta1", "namespaces", namespace, "iguaziotenantappservicesets", namespace}
+	_, err = s.kubeClientSet.Discovery().RESTClient().Patch(types.JSONPatchType).Body(body).AbsPath(absPath...).Do().Raw()
 	if err != nil {
-		s.logger.WarnWith("Failure during update of deployment", "resource_name", string(resource))
-		return errors.Wrap(err, "Failed updating deployment instance")
+		return errors.Wrap(err, "Failed to patch iguazio tenant app service sets")
 	}
+	return nil
+}
 
-	// if scaling up, make sure that all pods are in running state
-	if scaling != 0 {
-		s.logger.Info("Waiting for pods to be running")
-		if s.waitForServicePodsStatus(service, s.namespace, v1.PodRunning) != nil {
-			return errors.Wrap(err, "Failed while waiting for service pods status")
+func (s *AppResourceScaler) waitForServiceReadiness(namespace string, serviceName string) error {
+	s.logger.DebugWith("Waiting for service readiness", "serviceName", serviceName)
+	for {
+		resourcesList, err := s.GetResources()
+		if err != nil {
+			return errors.Wrap(err, "Failed to get ready services")
+		}
+		for _, resource := range resourcesList {
+			if string(resource) == serviceName {
+				s.logger.DebugWith("Service ready", "serviceName", serviceName)
+				return nil
+			}
 		}
-		s.logger.Info("All pods are running")
+		time.Sleep(5 * time.Second)
 	}
-	
 	return nil
 }
 
 func (s *AppResourceScaler) GetResources() ([]scaler_types.Resource, error) {
+	var iguazioTenantAppServicesSetMap map[string]interface{}
```

Comment:
This function doesn't `GetResources()` but `GetIguazioTenantAppAerviceAets()`
Also comment above about why needed

Reply:
Mmm `GetResources` is the name of the method in the resourceScaler interface that I'm implementing for app services, in which I practically get `IguazioTenantAppServiceSets`
https://github.com/v3io/scaler-types/blob/master/types.go#L37